scripts/Upgrade-PowerShell.ps1

```diff
@@ -167,62 +170,68 @@ Function Download-File($url, $path) {
     $client.DownloadFile($url, $path)
 }
 
-Function Clear-AutoLogon {
-    $reg_winlogon_path = "HKLM:\Software\Microsoft\Windows NT\CurrentVersion\Winlogon"
-    Write-Log -message "clearing auto logon registry properties"
-    Set-ItemProperty -Path $reg_winlogon_path -Name AutoAdminLogon -Value 0
-    Remove-ItemProperty -Path $reg_winlogon_path -Name DefaultUserName -ErrorAction SilentlyContinue
-    Remove-ItemProperty -Path $reg_winlogon_path -Name DefaultPassword -ErrorAction SilentlyContinue
+Function Set-AutoLogon {
+    param (
+        [switch] $Enable,
```

Comment:
I would probably just have the 1 switch, if set then enable AutoLogon, if not then remove AutoLogon. That way you don't need the ambiguous if case and Enable/Disable are opposites of each other.

Reply:
In that case, maybe a better way would be to make two separate functions - `Enable-AutoLogon` and `Disable-AutoLogon`?